### PR TITLE
RME / Oceanic : Workaround for high CPU Load in RXTX/nrjavaserial lib

### PR DESCRIPTION
**Pull request by kgoderis**
Signed-Off-By: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Hm, it seems the CPU load bug had been fixed by https://github.com/NeuronRobotics/nrjavaserial/pull/48. Is the workaround really still necessary?

**Comment by kgoderis** [comment]
Yes, it is. The nrjavaserial lib is still problematic for serial ports that send a constant stream of data. The work around in case of RME is to read à line of data every 5 seconds from this stream and throw away the rest, since the rme repetitively sends out the status of the  device on the serial port

File: addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialThingHandler.java [hunk 1]
```diff
@@ -94,33 +97,15 @@ public void writeString(String msg) {
...
-                    String port = (String) getConfig().get(PORT);
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "Error receiving data on serial port " + port + " : " + e.getMessage());
-                }
-                break;
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            /*
+             * See more details from
+             * https://github.com/NeuronRobotics/nrjavaserial/issues/22
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialThingHandler.java]: As you can see from my comment, this link didn't help me to understand the issue - and a `Thread.sleep(Long.MAX_VALUE);` will ring red alarm bells for anyone looking at that code.
Could you leave a comment in the sources here, why this workaround is still needed despite the fact that the referenced issue is closed and marked as solved?

Comment (kgoderis) [inline addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialThingHandler.java]: @kaikreuzer done.

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialThingHandler.java]: Thanks! I can see the comment in the RME code, but not here in Oceanic. Could you please copy it here as well?

File: addons/binding/org.openhab.binding.rme/src/main/java/org/openhab/binding/rme/handler/SerialThingHandler.java [hunk 2]
```diff
@@ -94,33 +97,18 @@ public void writeString(String msg) {
...
-                } catch (IOException e) {
-                    String port = (String) getConfig().get(PORT);
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "Error receiving data on serial port " + port + " : " + e.getMessage());
-                }
-                break;
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            /*
+             * The short select() timeout in the native code of the nrjavaserial lib does cause a high CPU load, despit
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.rme/src/main/java/org/openhab/binding/rme/handler/SerialThingHandler.java]: typo: despite

**Comment by kgoderis** [comment]
@kaikreuzer should be ok now.

**Pull request merged**